cfme/services/service_catalogs/ui.py

```diff
@@ -176,6 +176,8 @@ def order(self):
     if BZ(1513541, forced_streams=['5.9']).blocks:
         raise NotImplementedError("Service Order is broken - check BZ 1513541")
     view.submit_button.click()
+    if view.is_displayed and view.submit_button.is_displayed:
+        view.submit_button.click()
```

Comment:
**QUESTION**
Why are these checks needed?

Reply:
While running the whole module, there are some cases in which submit clicked too fast, and we get an error that last field is not filled, but it is. So I tried some check for field value is set and so on, this check will cover if submit is not successful

Reply:
Please add this comment to the code.

Reply:
Done